ledger/src/blockstore.rs

```diff
@@ -1500,19 +1503,43 @@ impl Blockstore {
 
     pub fn read_transaction_status(
         &self,
-        index: (Slot, Signature),
+        index: (Signature, Slot),
     ) -> Result<Option<RpcTransactionStatusMeta>> {
         self.transaction_status_cf.get(index)
     }
 
     pub fn write_transaction_status(
         &self,
-        index: (Slot, Signature),
+        index: (Signature, Slot),
         status: &RpcTransactionStatusMeta,
     ) -> Result<()> {
         self.transaction_status_cf.put(index, status)
     }
 
+    fn prune_transaction_status_cf_by_slot(
+        &self,
+        batch: &mut WriteBatch,
+        from_slot: Slot,
+        to_slot: Slot,
+    ) -> Result<bool> {
+        let mut results: Vec<bool> = vec![];
+        for slot in from_slot..to_slot {
+            let res = self
+                .get_slot_entries(slot, 0, None)?
+                .iter()
+                .cloned()
+                .flat_map(|entry| entry.transactions)
+                .map(|transaction| {
+                    batch
+                        .delete::<cf::TransactionStatus>((transaction.signatures[0], slot))
+                        .is_ok()
+                })
+                .all(|res| res);
+            results.push(res);
+        }
+        Ok(results.iter().all(|res| *res))
```

Comment:
From where this function's return value is being used above to determine `columns_empty`, it seems like this function should be returning whether or not the `TransactionStatus` column is empty, whereas this return value seems to indicate the success/failure of the computation.

Reply:
Ah, I was interpreting `columns_empty` as a marker of success, but I see that's slightly different.
What's the analog here? Should I just return `Ok(())`?

Reply:
In this revision, I'm just not handling the result of `prune_transaction_status_cf_by_slot()`

Reply:
hmmm that might be fine...depends how critical it is that `run_purge` accurately returns whether all the columns are empty. It seems like it's just used for an early exit condition in `purge_slots` @sunnygleason

Reply:
@sunnygleason , I could really use some advice on `compact_storage()` as well, if you're around!

Reply:
@CriesofCarrots it doesn't seem like the `compact_range` called on most column families by `compact_storage` will work for this new column as the keys you are deleting will be randomly distributed and not in a fixed range

Reply:
@carllin  That was my assumption as well. I tried removing this entirely https://github.com/solana-labs/solana/blob/c558db2a486226edb3ae767cb23846cc768a2d0b/ledger/src/blockstore.rs#L414 to not run compact_range on that column at all, but this test was failing here: https://github.com/solana-labs/solana/blob/b5dba770569402bfb2363f29d5e7cb3614add0ad/core/src/ledger_cleanup_service.rs#L166
I don't get why, unless any write to Blockstore does something to empty columns that needs to be compacted.